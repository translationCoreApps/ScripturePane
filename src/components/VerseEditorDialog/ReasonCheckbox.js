import React from 'react';
import PropTypes from 'prop-types';
import Checkbox from 'material-ui/Checkbox';

const styles = {
  root: {
    marginBottom: 16
  }
};

/**
 * Renders a reason checkbox
 * @param {string} reason - the reason for the edit
 * @param {string} label - the checkbox label
 * @param {string[]} selectedReasons - an array of selected reasons
 * @param {func} onCheck - callback when the checkbox is changed
 * @return {*}
 * @constructor
 */
class ReasonCheckbox extends React.Component {

  constructor(props) {
    super(props);
    this._handleCheck = this._handleCheck.bind(this);
  }

  _handleCheck(e, checked) {
    const {reason, onCheck} = this.props;
    onCheck(reason, checked);
  }

  render() {
    const {reason, label, selectedReasons} = this.props;
    return (
      <Checkbox style={styles.root}
                checked={selectedReasons.includes(reason)}
                onCheck={this._handleCheck}
                label={label}/>
    );
  }
}

ReasonCheckbox.propTypes = {
  reason: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  selectedReasons: PropTypes.arrayOf(PropTypes.string).isRequired,
  onCheck: PropTypes.func.isRequired
};

export default ReasonCheckbox;
