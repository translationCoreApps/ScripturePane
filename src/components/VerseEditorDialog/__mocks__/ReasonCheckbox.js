import React from 'react';

const mockCheckboxRender = jest.fn((props) => {
  const {reason, label, selectedReasons, onCheck} = props;
  return (
    <div>
      <input type="checkbox"
             onChange={onCheck}
             checked={selectedReasons.includes(reason)}
             name={reason}/> {label}
    </div>
  );
});

const mock = jest.fn().mockImplementation((props) => {
  return {
    render: mockCheckboxRender.bind(null, props)
  };
});

export default mock;
