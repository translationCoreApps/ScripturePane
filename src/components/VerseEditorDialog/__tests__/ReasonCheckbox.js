/* eslint-env jest */
import React from 'react';
import ReasonCheckbox from '../ReasonCheckbox';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {render} from 'enzyme';

describe('<ReasonCheckbox/>', () => {

  it('unselected', () => {
    const wrapper = render(
      <MuiThemeProvider>
        <ReasonCheckbox label="Label" onCheck={jest.fn()} reason="reason"/>
      </MuiThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
