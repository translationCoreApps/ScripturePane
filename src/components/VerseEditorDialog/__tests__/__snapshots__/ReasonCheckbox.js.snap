// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReasonCheckbox/> unselected 1`] = `
<div
  style="cursor:pointer;position:relative;overflow:visible;display:table;height:auto;width:100%;margin-bottom:16px;mui-prepared:"
>
  <input
    style="position:absolute;cursor:inherit;pointer-events:all;opacity:0;width:100%;height:100%;z-index:2;left:0;box-sizing:border-box;padding:0;margin:0;mui-prepared:"
    type="checkbox"
  />
  <div
    style="display:flex;width:100%;height:100%"
  >
    <div
      style="transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;float:left;position:relative;display:block;flex-shrink:0;width:24px;margin-right:16px;margin-left:0;height:24px;mui-prepared:"
    >
      <div>
        <svg
          style="display:inline-block;color:rgba(0, 0, 0, 0.87);fill:rgba(0, 0, 0, 0.87);height:24px;width:24px;user-select:none;transition:opacity 1000ms cubic-bezier(0.23, 1, 0.32, 1) 200ms;position:absolute;opacity:1;mui-prepared:"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <svg
          style="display:inline-block;color:rgba(0, 0, 0, 0.87);fill:#00bcd4;height:24px;width:24px;user-select:none;transition:opacity 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, transform 0ms cubic-bezier(0.23, 1, 0.32, 1) 450ms;position:absolute;opacity:0;transform:scale(0);transition-origin:50% 50%;mui-prepared:"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </div>
      <div />
      <div
        style="position:absolute;height:100%;width:100%;top:0;left:0;mui-prepared:"
      />
    </div>
    <label
      style="float:left;position:relative;display:block;width:calc(100% - 60px);line-height:24px;color:rgba(0, 0, 0, 0.87);font-family:Roboto, sans-serif;mui-prepared:"
    >
      Label
    </label>
  </div>
</div>
`;
